### ISSUE TITLE
multiple removes in ARRANGE breaks template

### ISSUE THREAD

**Issue by graybeal** [metadatacenter/cedar-project#1090]
In the training for DeiC, one participant found when he removed multiple terms using the ARRANGE feature under the VALUES tab, it prevented saving. I'll try to either duplicate or get more details.

**Comment by graybeal** [comment]
OK. This happened in this Template (video is at 5:28:00 of my DeIC 7/2 presentation), found by searching on Test_Johannes:
  https://cedar.metadatacenter.org/templates/edit/https://repo.metadatacenter.org/templates/dfb77fb6-3e59-4ca2-8adf-495f00af1db3

The image below was taken after opening the template, navigating to the field, clicking on the Values tab for the field called 'Type of vegetation', and removing the terms 'banana plantation' and 'carr', saving those changes, and saving the template.

![Screen Shot 2020-07-22 at 6.12.46 PM.png](https://images.zenhubusercontent.com/5cc9bdc26c1fd80bacaab82a/469c47af-6bbf-445c-bad0-77952710131f)

Now I've made a copy of that template, and I don't have the problem at all. 

When I go back to the original template, and click on Arrange for that field, the value list didn't come up for a long time the first time, but I couldn't duplicate that. When I went back to the original template a bit later and tried again, this time the Arrange changes worked even in the original template. 

So I don't have anything repeatable that I can show you (except for playing the video over and over), but I'd appreciate any insights before signing this off as 'can't duplicate'.

**Comment by graybeal** [comment]
@marcosmro is it possible that the model changes you pushed through a week ago have cleared up a problem (with the template, or with this feature) that could explain why it worked now?

**Comment by graybeal** [comment]
slacked this to Johanes: 

> When you are back in the world, either by waking up or returning from vacation, can you try duplicating the Arrange bug again? (Hint: The terms you deleted were 'banana plantation' and 'carr'.)  Thanks!

Not sure if he'll be back before their summer vacation is over (September).

**Comment by marcosmro** [comment]
I've been looking into this issue and the only possible cause that I can think of is that the Template Editor was more recent than the validation library at the moment of that error.

Last April, when implementing the integrated search for the embeddable editor, I added a 'source' field to each 'action' to be consistent with our representation of other terms in the value constraints. I also updated the validation library to require that field, and patched manually some resources in the system that were using the actions capability. 

If, at the moment of the error, Production was using a validation library older than the template editor, the validator wouldn't accept the 'source' field as valid, and would throw a bad request. 

It's possible that for some UI improvement that needed to be released with urgency, we pulled the most recent code of the template editor without pulling the other repos.

Not sure 100% if that was the cause, but that's my only hypothesis.

For more info, here are the commits with those changes:
- Template Editor: https://github.com/metadatacenter/cedar-template-editor/commit/4337e7b6c7f6cab9b05db05327418fbdf3cf5084
- Validation library: https://github.com/metadatacenter/cedar-model-validation-library/commit/41f92adda3098b8eb534ac15941cfe58c6d6750b
